sigmadsp/hardware/dsp.py

```diff
@@ -101,6 +109,13 @@ def parse_config(self):
         except (KeyError, TypeError):
             logger.info("No DSP pin definitions were found in the configuration file.")
 
+        self.type = self.config["dsp"]["type"]
+        self.protocol = self.config["dsp"].get("protocol", "spi")
+
+        if self.protocol == "i2c":
+            self.i2c_address = self.config["dsp"].get("i2c_address", 0x38)
+            self.i2c_bus = self.config["dsp"].get("i2c_bus", 1)
```

Comment:
Please add some log message for when a default value is chosen (so, no value found in config).

Reply:
hm, I think it might be better to actually require specifying the bus / address in the configuration

Reply:
Yes, I agree.